packages/docusaurus-mdx-loader/src/remark/transformLinks/index.ts

```diff
@@ -54,23 +65,43 @@ function toAssetRequireNode(
   const hash = parsedUrl.hash ?? '';
   const search = parsedUrl.search ?? '';
 
-  const href = `require('${
+  const requireString = `${
     // A hack to stop Webpack from using its built-in loader to parse JSON
     path.extname(relativeAssetPath) === '.json'
       ? `${relativeAssetPath.replace('.json', '.raw')}!=`
       : ''
-  }${inlineMarkdownLinkFileLoader}${
-    escapePath(relativeAssetPath) + search
-  }').default${hash ? ` + '${hash}'` : ''}`;
-  const children = stringifyContent(node);
-  const title = node.title ? ` title="${escapeHtml(node.title)}"` : '';
-
-  const jsxNode: Literal = {
-    type: 'jsx',
-    value: `<a target="_blank" href={${href}}${title}>${children}</a>`,
-  };
+  }${inlineMarkdownLinkFileLoader}${escapePath(relativeAssetPath) + search}`;
+
+  attributes.push({
+    type: 'mdxJsxAttribute',
+    name: 'target',
+    value: '_blank',
+  });
+
+  attributes.push({
+    type: 'mdxJsxAttribute',
+    name: 'href',
+    value: assetRequireAttributeValue(requireString, hash),
+  });
+
+  if (node.title) {
+    attributes.push({
+      type: 'mdxJsxAttribute',
+      name: 'title',
+      value: escapeHtml(node.title),
+    });
+  }
+
+  const {children} = node;
+
+  Object.keys(jsxNode).forEach(
+    (key) => delete jsxNode[key as keyof typeof jsxNode],
+  );
 
-  parent.children.splice(index, 1, jsxNode);
+  jsxNode.type = 'mdxJsxFlowElement';
```

Comment:
are you sure this type is correct? i think this should be mdxJsxTextElement

Reply:
Same as:

````markdown
```js
console.log(1)
```

^-- flow

`text` <-- text
````

---

The AST isn’t only about the result.
Also depends on where in the AST you do things.
That the final result is equivalent, doesn’t mean the AST is the same!

Reply:
Thanks @wooorm @armano2 , going to use mdxJsxTextElement then 👍

Reply:
I'm assuming it's the same for images, although using images inside a paragraph is less common it remains possible 🤷‍♂️

Reply:
> Thanks @wooorm @armano2 , going to use mdxJsxTextElement then 👍

i was not 100% sure if this was correct that's why i asked question :)